neural_compressor/torch/quantization/load_entry.py

```diff
@@ -25,43 +25,85 @@
+        device (str, optional): 'cpu', 'hpu' or 'cuda'. specify the device the model will be loaded to.
+        kwargs (remaining dictionary of keyword arguments, optional):
+            remaining dictionary of keyword arguments for loading huggingface models.
+            Will be passed to the huggingface model's `__init__` method, such as 'trust_remote_code', 'revision'.
     Returns:
         The quantized model
     """
-    from neural_compressor.common.base_config import ConfigRegistry
+    # TODO: When loading WOQ model, use different WeightOnlyLinear module according to device.
+    if format == LoadFormat.DEFAULT.value:
+        from neural_compressor.common.base_config import ConfigRegistry
 
-    qconfig_file_path = os.path.join(os.path.abspath(os.path.expanduser(output_dir)), "qconfig.json")
-    with open(qconfig_file_path, "r") as f:
-        per_op_qconfig = json.load(f)
+        qconfig_file_path = os.path.join(os.path.abspath(os.path.expanduser(model_name_or_path)), "qconfig.json")
+        with open(qconfig_file_path, "r") as f:
+            per_op_qconfig = json.load(f)
 
-    if " " in per_op_qconfig.keys():  # ipex qconfig format: {' ': {'q_op_infos': {'0': {'op_type': ...
-        from neural_compressor.torch.algorithms.static_quant import load
+        if " " in per_op_qconfig.keys():  # ipex qconfig format: {' ': {'q_op_infos': {'0': {'op_type': ...
+            from neural_compressor.torch.algorithms.static_quant import load
 
-        return load(output_dir)
-    else:
-        config_mapping = load_config_mapping(qconfig_file_path, ConfigRegistry.get_all_configs()["torch"])
-        # select load function
-        config_object = config_mapping[next(iter(config_mapping))]
-        if isinstance(config_object, (RTNConfig, GPTQConfig, AWQConfig, TEQConfig, AutoRoundConfig)):  # WOQ
-            from neural_compressor.torch.algorithms.weight_only.save_load import load
+            return load(model_name_or_path)
+        else:
+            config_mapping = load_config_mapping(qconfig_file_path, ConfigRegistry.get_all_configs()["torch"])
+            # select load function
+            config_object = config_mapping[next(iter(config_mapping))]
+
+            if isinstance(config_object, (RTNConfig, GPTQConfig, AWQConfig, TEQConfig, AutoRoundConfig)):  # WOQ
+                from neural_compressor.torch.algorithms.weight_only.save_load import load
 
-            return load(output_dir)
+                return load(model_name_or_path, original_model, format=LoadFormat.DEFAULT)
 
-        model.qconfig = config_mapping
-        if isinstance(config_object, FP8Config):  # FP8
-            from neural_compressor.torch.algorithms.habana_fp8 import load
+            original_model.qconfig = config_mapping
+            if isinstance(config_object, FP8Config):  # FP8
+                from neural_compressor.torch.algorithms.habana_fp8 import load
```

Comment:
nit: there are several `load` functions, maybe using the Google style imports makes it easier to understand.

```python
import neural_compressor.torch.algorithms.habana_fp8 as fp8
fp8.load(...)
```

Reply:
update in https://github.com/intel/neural-compressor/pull/1826/commits/a33c766e8865f009ab02fa4ccd24cedd85574113